modules/apps/frontend-js/frontend-js-importmaps-extender/src/main/java/com/liferay/frontend/js/importmaps/extender/internal/servlet/taglib/JSImportMapsExtenderTopHeadDynamicInclude.java

```diff
@@ -192,33 +177,63 @@ private synchronized void _rebuildImportMaps() {
 			() -> {
 				JSONObject scopesJSONObject = _jsonFactory.createJSONObject();
 
-				for (Map.Entry<String, JSONObject> entry :
-						_scopedImportMapJSONObjects.entrySet()) {
+				for (JSImportMapsContributor jsImportMapsContributor :
+						_jsImportMapsContributors.keySet()) {
 
-					scopesJSONObject.put(entry.getKey(), entry.getValue());
+					if (jsImportMapsContributor.getScope() == null) {
+						continue;
+					}
+
+					String scope = jsImportMapsContributor.getScope();
+
+					if (scopesJSONObject.has(scope)) {
+						throw new IllegalStateException(
+							"Found multiple JS Import Maps Contributors for " +
+								"the same scope " + scope);
+					}
+
+					scopesJSONObject.put(
+						scope,
+						jsImportMapsContributor.getImportMapsJSONObject(
+							locale));
```

Comment:
What happens here exactly? Will it just emit this error and then override the existing scope? Should this be if/else?

I would think in the case where a second scope is added, it shouldn't override the existing one.

Reply:
Nope, if the exception is thrown the execution stops. So, once a scope is set it remains forever unless it is uninstalled.